platform/darwin/src/MGLStyleValue.h

```diff
@@ -135,77 +174,261 @@ MGL_EXPORT
 
 @end
 
+/**
+ An `MGLStyleFunction` is a is an abstract superclass for functions that are 
+ defined by an `MGLCameraStyleFunction`, `MGLSourceStyleFunction`, or
+ `MGLCompositeStyleFunction` object.
+ 
+ Do not create instances of this class directly, and do not create your own
+ subclasses of this class. Instead, use one of the class factory methods in
+ `MGLStyleValue` to create instances of the following concrete subclasses:
+ `MGLCameraStyleFunction`, `MGLSourceStyleFunction`, and
+ `MGLCompositeStyleFunction`.
+
+ The `MGLStyleFunction` class takes a generic parameter `T` that indicates the
+ Foundation class being wrapped by this class.
+ */
 MGL_EXPORT
-@interface MGLCameraStyleFunction<T> : MGLStyleValue<T>
-
-// TODO: API docs
-+ (instancetype)functionWithInterpolationMode:(MGLInterpolationMode)interpolationMode stops:(NS_DICTIONARY_OF(id, MGLStyleValue<T> *) *)stops options:(nullable NS_DICTIONARY_OF(MGLStyleFunctionOption, id) *)options;
+@interface MGLStyleFunction<T> : MGLStyleValue<T>
 
-// TODO: API docs
-- (instancetype)initWithInterpolationMode:(MGLInterpolationMode)interpolationMode stops:(NS_DICTIONARY_OF(id, MGLStyleValue<T> *) *)stops options:(nullable NS_DICTIONARY_OF(MGLStyleFunctionOption, id) *)options NS_DESIGNATED_INITIALIZER;
+#pragma mark Accessing the Parameters of a Function
 
-// TODO: API docs
+/**
+ The modes used to interpolate property values between map zoom level changes or
+ over a range of feature attribute values.
+ */
 @property (nonatomic) MGLInterpolationMode interpolationMode;
 
-// TODO: API docs
-@property (nonatomic, copy) NS_DICTIONARY_OF(id, MGLStyleValue<T> *) *stops;
+/**
+ A dictionary associating zoom levels with style values.
+ */
+@property (nonatomic, copy) NSDictionary *stops;
 
-// TODO: API docs
+/**
+ The exponential interpolation base of the function’s interpolation curve.
+ */
```

Comment:
Document the fact that `interpolationBase` may not apply to all style functions.